src/frontend/Semantic_check.ml

```diff
@@ -1926,7 +1934,5 @@ let semantic_check_program
   Validate.(
     ok mk_typed_prog |> apply_to ufb |> apply_to udb |> apply_to utdb
     |> apply_to upb |> apply_to utpb |> apply_to umb |> apply_to ugb
-    |> check_correctness_invariant_validate
-    |> get_with
-         ~with_ok:(fun ok -> Result.Ok ok)
-         ~with_errors:(fun errs -> Result.Error errs))
+    |> check_correctness_invariant_validate)
+    |> attach_warnings
```

Comment:
I believe the way you would get the correct return type here would be 

```ocaml
|> get_with
      ~with_ok:(fun ok -> Result.Ok (attach_warnings ok))
      ~with_errors:(fun errs -> Result.Error errs))
```

Reply:
Thanks!